apps/console/src/features/roles/components/wizard/role-permission.tsx

```diff
@@ -69,32 +69,25 @@ export const PermissionList: FunctionComponent<PermissionListProp> = (props: Per
      */
     useEffect(() => {
         if (isRole && roleObject) {
-            getPermissionsForRole(roleObject.id)
-                .then(response => {
-                    if (response.status === 200 && response.data instanceof Array) {
-                        const permissionsArray: Permission[] = [];
-                        const permissions = response.data;
-                        permissions.forEach(permission => {
-                            permissionsArray.push({
-                                fullPath: permission,
-                                id: permission,
-                                isChecked: false,
-                                isExpanded: true,
-                                isPartiallyChecked: false,
-                                name: permission
-                            })
-                        });
-                        setAvailablePermissionsInRole(permissionsArray);
-                        getAllPermissions();
-                    }
-                })
-                .catch(() => {
-                    //Handle Role Retrieval Properly
-                })
+            const permissionsArray: Permission[] = [];
+            const permissions = roleObject.permissions;
+            permissions?.forEach(permission => {
+                permissionsArray.push({
+                    fullPath: permission,
+                    id: permission,
+                    isChecked: false,
+                    isExpanded: true,
+                    isPartiallyChecked: false,
+                    name: permission
+                });
+            });
+            setAvailablePermissionsInRole(permissionsArray);
+            getAllPermissions();
+
         } else {
             getAllPermissions();
         }
-    }, [availablePermissionsInRole.toString()]);
+    }, [ availablePermissionsInRole.toString() ]);
```

Comment:
Why are we making this string here ?

Reply:
We'll address this separately in another PR. And merge this.

Reply:
The relevant issue is tracked here: https://github.com/wso2/product-is/issues/10157